cart/domain/cart/cart.go

```diff
@@ -461,6 +479,15 @@ func (c Cart) SumTotalDiscountAmount() domain.Price {
 	return price
 }
 
+// SumTotalDiscountWithGiftCardsAmount – returns sum price of total discounts with applied gift cards
+func (c Cart) SumTotalDiscountWithGiftCardsAmount() domain.Price {
```

Comment:
Semantically I find it hard to get the intent:

Discount is something that reduced the price that need to be payed.

Giftcard is a way of paying something.

Why would we provide this in the cart domain?

If for display reasons you want to sum that up - you are free to do it in the interface layer..

Imho hard to argument that it belongs to a typical thing you do in the cart domain (summing a discount and a gifcart payment)

Reply:
From concept side we thought we would sum up all "discounts" (cart rules, coupons and gift cards) and display the "discounts" to customer. Yes this is probably something very specific, in the old days we could've done this in the template, but in the graphql "world" this should be in the interface layer.